### ISSUE TITLE
[Typescript] Mismatch between type definitions and effective module export policy

### ISSUE THREAD

**Issue by jsamr** [styled-components/polished#276]
- `polished` version: 1.9.0
- `JSS-in_CSS` library and version: *not relevant*
- Any relevant JS library and version: Typescript 2.5.3 or 2.6

**Mixin/Helper/Shortand you were using and how you were using it:**

```typescript
import readableColor from 'polished/lib/color/readableColor'

console.info('Readable color function:', readableColor)
```

**What you are seeing**: 

`Readable color function: undefined` printed in the console

**What you expected to see:** 
 
`Readable color function: function...` printed in the console

As per the type definitions: 
```typescript
declare function curriedReadableColor(color: string): string

export default curriedReadableColor;
```
`curriedReadableColor` should be exported as es module `default`.

**Reproduction**: [minimal working example](https://github.com/sveinburne/polished-export-bug)

**Inspection and possible fix**
The bug comes from the transpiled source: 
```javascript
// src: node_modules/polished/lib/color/readableColor.js
// ... line 50
exports.default = curriedReadableColor;
module.exports = exports['default'];
```
The last line overwrite `module.exports.default` by direct assignment of the function. If this is the intended building behavior, you should rewrite typescript definitions with `export = curriedReadableColor` ; you could do the same as [lodash does for function direct exports](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/lodash/filter.d.ts).

**Remark** The example given at the beginning will not raise problems with any building solution supporting synthetic default exports such as babel (with `transform-es2015-modules-commonjs`) or fusebox. 

**Typescript compiler configuration file**
(note that `tsc` is directly used and sources are run with nodejs 6.12)
```json
{
  "compilerOptions": {
    "target": "es5",
    "module": "CommonJS",
    "moduleResolution": "node",
    "allowSyntheticDefaultImports": false,
    "lib": ["es6", "es7", "dom"],
    "strict": true
  }
}
```

**Comment by oyvindberg** [comment]
This bit me as well. Possible workaround:
```typescript
const shade: (percentage: number, color: string) => string = require('polished/lib/color/shade')
```

Any idea where the mismatch comes from? Does this work with flow?

**Comment by bhough** [comment]
@oyvindberg The typings haven't been removed in the current version. This bug will be addressed in 2.0, but here is the info as to why you are seeing that issue: https://github.com/styled-components/polished/pull/295#issuecomment-364507179

**Comment by bhough** [comment]
Addressed by #321 and merged into `version-2`.
